File: src/main/resources/frontend/commando-app/src/app/models/month-year.ts
/*
 * Copyright (c) 2020. Energy, Inc.
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *          http://www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 */

export class MonthYearRange {
  public start:MonthYear = null;
  public end:MonthYear = null;
}

export class MonthYear {
  public month:number;
  public year:number;
  public selected:boolean = false;

  /***
   * Initializes the Month year using the specified date.
   * @param date
   * @param selected
   */
  constructor(date:Date = new Date(), selected: boolean = false) {
    this.month = date.getMonth();
    this.year = date.getFullYear();
    this.selected = selected;
  }

  public compare(o:MonthYear):number{
    if (o == null) return 1;
    let thisValue = (this.year *100) + this.month;
    let thatValue = (o.year * 100) + o.month;
    if (thisValue < thatValue) return -1;
    else if (thisValue == thatValue) return 0;
    else return 1;
  }

  public isInBetween(startDate:MonthYear, endDate:MonthYear):boolean{
    return  (this.compare(startDate) >= 0) && (this.compare(endDate) <= 0)
  }
}

File: src/main/java/com/ted/commando/model/MonthYear.java
/*
 * Copyright (c) 2020. Energy, Inc.
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *          http://www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 */

package com.ted.commando.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;


@JsonIgnoreProperties(ignoreUnknown = true)
public class MonthYear {
    private int month = 0;
    private int year = 0;
    private boolean selected = false;

    public int getMonth() {
        return month;
    }

    public void setMonth(int month) {
        this.month = month;
    }

    public int getYear() {
        return year;
    }

    public void setYear(int year) {
        this.year = year;
    }

    public boolean isSelected() {
        return selected;
    }

    public void setSelected(boolean selected) {
        this.selected = selected;
    }

    @Override
    public String toString() {
        return "MonthYear{" +
                "month=" + month +
                ", year=" + year +
                ", selected=" + selected +
                '}';
    }
}
